src/hdf5/spire_2ioda.py

```diff
@@ -0,0 +1,281 @@
+        varDict[iodavar]['valKey'] = iodavar, obsValName
+        varDict[iodavar]['errKey'] = iodavar, obsErrName
+        varDict[iodavar]['qcKey'] = iodavar, qcName
+        varAttrs[iodavar, obsValName]['coordinates'] = 'longitude latitude'
+        varAttrs[iodavar, obsErrName]['coordinates'] = 'longitude latitude'
+        varAttrs[iodavar, qcName]['coordinates'] = 'longitude latitude'
+        varAttrs[iodavar, obsValName]['units'] = obsvars_units[n]
+        varAttrs[iodavar, obsErrName]['units'] = obsvars_units[n]
+
+    # Set units of the MetaData variables and all _FillValues.
+    for key in meta_keys:
+        dtypestr = MetaDataKeyList[meta_keys.index(key)][1]
+        if MetaDataKeyList[meta_keys.index(key)][2]:
+            varAttrs[(key, metaDataName)]['units'] = MetaDataKeyList[meta_keys.index(key)][2]
+        varAttrs[(key, metaDataName)]['_FillValue'] = missing_vals[dtypestr]
+        ioda_data[(key, metaDataName)] = np.array(obs_data[key], dtype=dtypes[dtypestr])
+
+    # Transfer from the 1-D data vectors and ensure output data (ioda_data) types using numpy.
+    for n, iodavar in enumerate(obsvars):
+        ioda_data[(iodavar, obsValName)] = np.array(obs_data[iodavar], dtype=np.float32)
+#        ioda_data[(iodavar, obsErrName)] = np.full(ntotal, obserrlist[n], dtype=np.float32)
+        ioda_data[(iodavar, qcName)] = np.full(ntotal, 2, dtype=np.int32)
+
+    # setup the IODA writer
+    writer = iconv.IodaWriter(args.output_file, MetaDataKeyList, DimDict)
+    # write everything out
+    writer.BuildIoda(ioda_data, VarDims, varAttrs, GlobalAttrs)
+
+
+def get_data_from_file(afile, col_names, file_name):
+
+    # Get instrument reference
+    instrument_ref = afile.attrs['tx_id'].decode('UTF-8') 
+
+    # Pull each data type (variable) and create a list
+    latitude = [afile['sp_lat'][ii] for ii in range(len(afile['sp_lat']))]
+    longitude = [afile['sp_lon'][ii] for ii in range(len(afile['sp_lon']))]
+    dateTime = [int(afile['sample_time'][ii]) for ii in range(len(afile['sample_time']))]  # datetime with different ref time
+    windSpeedAt10M = [afile['wind'][ii] for ii in range(len(afile['wind']))]
+    sensorIdentification = [instrument_ref]*len(latitude)
+
+    # Make a column to have a constant elevation for the "station"
+    height = [10]*len(dateTime)
+
+    # Make a list of lists to feed into dataframe
+    data_lists = list(zip(latitude, longitude, dateTime, sensorIdentification,
+                          height, windSpeedAt10M))
+
+    # All observation data for this file to append to the master dataframe
+    obs_data_append = pd.DataFrame(data_lists, columns=col_names)
+    return obs_data_append
+
+
+def adjust_dateTime(obs_DF, dat_ref):
+    obs_DF['dateTime'] = obs_DF['dateTime']+int(dat_ref)
+    return obs_DF
+
+def adjust_longitude(obs_DF):
+    obs_DF['longitude'] = (obs_DF['longitude']+int(180)) % 360 - int(180)
+    return obs_DF
+
+
+def quality_control(obs_data):
```

Comment:
Not sure how wind speed could be a negative value.  Am I mis-interpreting the information?  Is the velocity in this data feed a value of wind `towards` the instrument and `away` from the instrument such as in doppler radar?  That's the only reason a negative value should be possible, right?

Reply:
The speed should definitely be positive. Negative would only be if it's an error, but correct, a range of `[-100, 100]' is not needed. '[0, 100]' will suffice.